src/block/get.js

```diff
@@ -0,0 +1,99 @@
+/* eslint-env serviceworker, browser */
+/* global Response */
+
+import { CID } from 'multiformats/cid'
+import { base58btc } from 'multiformats/bases/base58'
+
+import { BlockNotFoundError, BaseNotFoundError } from '../errors.js'
+
+/**
+ * @typedef {import('../env').Env} Env
+ */
+
+/**
+ * Handle block get request
+ *
+ * @param {Request} request
+ * @param {Env} env
+ * @param {import('../index').Ctx} ctx
+ */
+export async function blockGet (request, env, ctx) {
+  // Get cached block if exists
+  const cache = caches.default
+  let res = await cache.match(request)
+  if (res) {
+    return res
+  }
+
+  const multihashOrCid = request.params.multihash
+
+  // Permanently redirect to multihash if cid provided
+  // Note that CIDv0 and multihash encoded as b58btc will be the same
+  const multihashByCidValue = getMultihashFromCidValue(multihashOrCid)
+  if (multihashByCidValue && multihashByCidValue !== multihashOrCid) {
+    return Response.redirect(
+      request.url.replace(multihashOrCid, multihashByCidValue),
+      301
+    )
+  }
+
+  const multihash = multihashOrCid
+  const key = await toBase58btc(multihash, env.bases)
+
+  const r2Object = await env.BLOCKSTORE.get(key)
+  if (r2Object) {
+    res = new Response(r2Object.body)
```

Comment:
whats our cache strategy here ? there no header being added to the response to control it

Reply:
It is immutable and we want to cache it permanently all the time. So, no need at this point to parse header in request.
Should we send cache-control immutable header in response though?

Reply:
>  is it cached "permanently" until the LRU limit is reached ?

Yes, we can add immutable to Cache-Control as I suggested to hint clients. But, I think there is no other header property per https://developers.cloudflare.com/cache/about/cache-control#cache-control-directives and https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control

Let me know if you had any other in mind, I will add immutable